underfs/s3a/src/main/java/alluxio/underfs/s3a/S3AUnderFileSystem.java

```diff
@@ -208,17 +207,49 @@ public static S3AUnderFileSystem createInstance(AlluxioURI uri,
       clientConf.setSignerOverride(conf.get(PropertyKey.UNDERFS_S3_SIGNER_ALGORITHM));
     }
 
-    AmazonS3ClientBuilder clientBuilder = AmazonS3Client.builder()
-        .withCredentials(credentials)
+    AwsClientBuilder.EndpointConfiguration endpointConfiguration
+        = createEndpointConfiguration(conf, clientConf);
```

Comment:
Do we expect creating different endpoint configuration for each Amazon S3 client?
why not put create endpoint configuration into createAmazonS3?

Reply:
In the other scenarios, the client would be created multiple times, and not necessary to create endpoint multiple times. The purpose is trying to share some common objects.